hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/containermanager/linux/runtime/DockerLinuxContainerRuntime.java

```diff
@@ -343,6 +350,11 @@ public void initialize(Configuration conf, Context nmContext)
         YarnConfiguration.NM_DOCKER_IMAGE_NAME, "");
     defaultImageUpdate = conf.getBoolean(
         YarnConfiguration.NM_DOCKER_IMAGE_UPDATE, false);
+    Arrays.stream(conf.get(YarnConfiguration.NM_DOCKER_ALLOWED_IMAGES, "").split(","))
+        .forEach( image-> {
+            String[] ImageNameAndVersion= image.split("=");
+            allowedImages.put(ImageNameAndVersion[0], ImageNameAndVersion[1]);
```

Comment:
this can fail with IndexOutOfBound exception if the property is badly configure. Can you please check the array size and provided a meaningful error ?

Reply:
I have updated the review to add a validation on the allowed images entries.